server/src/main/java/org/elasticsearch/search/aggregations/pipeline/AbstractPipelineAggregationBuilder.java

```diff
@@ -102,6 +104,26 @@ public final PipelineAggregator create() throws IOException {
     public void doValidate(AggregatorFactory<?> parent, Collection<AggregationBuilder> factories,
             Collection<PipelineAggregationBuilder> pipelineAggregatorFactories) {
     }
+    
+    /**
+     * Validates pipeline aggregations that need sequentially ordered data.
+     */
+    public static void validateSequentiallyOrderedParentAggs(AggregatorFactory<?> parent, String type, String name) {
+        if (parent instanceof HistogramAggregatorFactory) {
+            HistogramAggregatorFactory histoParent = (HistogramAggregatorFactory) parent;
+            if (histoParent.minDocCount() != 0) {
+                throw new IllegalStateException("parent histogram of " + type + " aggregation [" + name + "] must have min_doc_count of 0");
+            }
+        } else if (parent instanceof DateHistogramAggregatorFactory) {
+            DateHistogramAggregatorFactory histoParent = (DateHistogramAggregatorFactory) parent;
+            if (histoParent.minDocCount() != 0) {
+                throw new IllegalStateException("parent histogram of " + type + " aggregation [" + name + "] must have min_doc_count of 0");
+            }
+        } else {
```

Comment:
Ah sorry, this is my fault I wasn't clear enough.  We'll still need to make sure the parent agg is `HistogramAggregatorFactory`, `DateHistogramAggregatorFactory` or `AutoDateHistogramAggregatorFactory`.  It's just that the `AutoDateHistogramAggregatorFactory` case won't have any additional validation like checking min_doc_count.

So it'll just be an empty if clause.  We still need to make sure it's one of the three classes though, otherwise the exception below will throw like the original issue.